packages/cache-core/src/strategies/memory.ts

```diff
@@ -46,11 +46,7 @@ export class MemoryCache implements CacheStrategy {
   async revalidateTag(tag: string): Promise<void> {
     for (const cacheKey of [...mapCache.keys()]) {
       const pageData: CacheEntry = JSON.parse(mapCache.get(cacheKey))
-      if (
-        pageData?.tags?.includes(tag) ||
-        (pageData?.value?.kind === 'PAGE' &&
-          pageData.value?.headers?.[NEXT_CACHE_TAGS_HEADER]?.toString()?.split(',').includes(tag))
-      ) {
+      if (pageData?.tags?.includes(tag) || checkHeaderTags(pageData.value, tag)) {
```

Comment:
we should update tests for new logic

Reply:
Yeah, I am going to update it. I will prepare separate PR and test it in different cases